delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/batch/BatchFiles.scala

```diff
@@ -0,0 +1,81 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.storage.files.batch
+
+import cats.data.NonEmptyList
+import cats.effect.IO
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.delta.kernel.Logger
+import ch.epfl.bluebrain.nexus.delta.kernel.kamon.KamonMetricComponent
+import ch.epfl.bluebrain.nexus.delta.kernel.utils.UUIDF
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.Files.entityType
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model.FileCommand._
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model.FileRejection.CopyRejection
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model._
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.routes.CopyFileSource
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.{FetchFileStorage, FileResource}
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.model.StorageType
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.operations.StorageFileRejection.CopyFileRejection
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.sdk.identities.model.Caller
+import ch.epfl.bluebrain.nexus.delta.sdk.implicits._
+import ch.epfl.bluebrain.nexus.delta.sdk.projects.FetchContext
+import ch.epfl.bluebrain.nexus.delta.sdk.projects.model.ProjectContext
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.ResourceRef
+
+trait BatchFiles {
+  def copyFiles(
+      source: CopyFileSource,
+      dest: CopyFileDestination
+  )(implicit c: Caller): IO[NonEmptyList[FileResource]]
+}
+
+object BatchFiles {
+  def mk(
+      fetchFileStorage: FetchFileStorage,
+      fetchContext: FetchContext[FileRejection],
+      evalFileCommand: CreateFile => IO[FileResource],
+      batchCopy: BatchCopy
+  )(implicit uuidF: UUIDF): BatchFiles = new BatchFiles {
+
+    private val logger = Logger[BatchFiles]
+
+    implicit private val kamonComponent: KamonMetricComponent = KamonMetricComponent(entityType.value)
+
+    override def copyFiles(source: CopyFileSource, dest: CopyFileDestination)(implicit
+        c: Caller
+    ): IO[NonEmptyList[FileResource]] = {
+      for {
+        pc                            <- fetchContext.onCreate(dest.project)
+        (destStorageRef, destStorage) <- fetchFileStorage.fetchAndValidateActiveStorage(dest.storage, dest.project, pc)
+        destFilesAttributes           <- batchCopy.copyFiles(source, destStorage).adaptError { case e: CopyFileRejection =>
+                                           CopyRejection(source.project, dest.project, destStorage.id, e)
+                                         }
+        fileResources                 <- evalCreateCommands(pc, dest, destStorageRef, destStorage.tpe, destFilesAttributes)
```

Comment:
I realise that we are "evaling" create "commands", but maybe it's better to just say we are creating the files? Just as a more domain-oriented reading of what is happening

Reply:
The actual files already exist so maybe create file resources 👍